src/QMCHamiltonians/ECPComponentBuilder.2.cpp

```diff
@@ -109,6 +131,10 @@ void ECPComponentBuilder::buildSemiLocalAndLocal(std::vector<xmlNodePtr>& semiPt
     Llocal = Lmax;
     app_log() << "    Only one vps is found. Set the local component=" << Lmax << std::endl;
   }
+
+  if (angListSO.size()!=nso)
+    APP_ABORT("Error. npots-so does not match number of spin-orbit channels.");
```

Comment:
The error message could be improved by printing the values of npots-so and the angListSO size.

Reply:
Done.